--- a/include/kernel-defaults.mk
+++ b/include/kernel-defaults.mk
@@ -111,6 +111,9 @@ define Kernel/Configure/Default
 		cp $(LINUX_DIR)/.config.set $(LINUX_DIR)/.config.prev; \
 	}
+	$(_SINGLE)$(KERNEL_MAKE) olddefconfig
+	cp $(LINUX_DIR)/.config $(LINUX_DIR)/.config.set
+	cp $(LINUX_DIR)/.config $(LINUX_DIR)/.config.prev
 	$(_SINGLE) [ -d $(LINUX_DIR)/user_headers ] || $(KERNEL_MAKE) INSTALL_HDR_PATH=$(LINUX_DIR)/user_headers headers_install
 	grep '=[ym]' $(LINUX_DIR)/.config.set | LC_ALL=C sort | mkhash md5 > $(LINUX_DIR)/.vermagic
 endef
